src/main/java/ui/UI.java

```diff
@@ -540,4 +544,27 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    /**
+     * Warns user if the Java runtime version is lower than HT's requirement
+     */
+    private void warnIfJavaVersionOutdated() {
+        try {
+            JavaVersion runtimeVersion = JavaVersion.fromString(System.getProperty("java.runtime.version"));
+            JavaVersion requiredVersion = JavaVersion.fromString(REQUIRED_VERSION);
+
+            if (JavaVersion.isJavaVersionTooLow(runtimeVersion, requiredVersion)) {
+                showJavaVersionOutdatedWarning(runtimeVersion, requiredVersion);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Java Version string failed to be parsed.", e);
```

Comment:
I'm not sure if this is the best way to handle this error. Shouldn't there be a more visible indication if this happens?